### PR TITLE
Update actions, run fmt, run clippy for baseline.

### PR DESCRIPTION
**Pull request by waywardmonkeys**

### CODE REVIEW

**Review by Wumpf — CHANGES_REQUESTED** [review]
the advantage of `actions-rs/clippy-check@v1` is that it marks up errors in the PR diff, so I'd like to stick with that or similar

**Comment by Wumpf** [comment]
`cargo test` can't actually run on CI since it requires a proper device. Doing this with a software rasterizer should be possible, but I believe we had some issues with queries over on wgpu. For the time being I'd simply drop that from ci

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -11,12 +11,12 @@ env:
...
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
-    - run: rustup component add clippy
-    - uses: actions-rs/clippy-check@v1
-      with:
-        token: ${{ secrets.GITHUB_TOKEN }}
-        args: --all-features
+    - uses: actions/checkout@v4
+    - uses: dtolnay/rust-toolchain@stable
```
Comment (Wumpf) [inline .github/workflows/rust.yml:18]: why though, rust is already installed

Comment (waywardmonkeys) [reply .github/workflows/rust.yml:18]: It won't necessarily be the current version, just whenever GitHub updates their images.

**Comment by waywardmonkeys** [comment]
> the advantage of `actions-rs/clippy-check@v1` is that it marks up errors in the PR diff, so I'd like to stick with that or similar

The problem is that the `actions-rs` stuff is all unmaintained for a long time...

**Comment by Wumpf** [comment]
hmm sucks, do you happen to know if there's something more maintained that also has this?

**Comment by Imberflur** [comment]
> > the advantage of `actions-rs/clippy-check@v1` is that it marks up errors in the PR diff, so I'd like to stick with that or similar
> 
> The problem is that the `actions-rs` stuff is all unmaintained for a long time...

The documentation also mentions that these annotations don't work when the PR is from a fork... https://github.com/actions-rs/clippy-check#limitations

Someone in the issue linked there pointed to https://github.com/giraffate/clippy-action which I think employ a different method for displaying the errors.

**Comment by waywardmonkeys** [comment]
> hmm sucks, do you happen to know if there's something more maintained that also has this?

I don't at the moment.

**Review by Wumpf — APPROVED** [review]
alright fair enough, makes sense!
Thanks for putting these in

**Pull request merged**